src/libraries/ExcelConverter/ParsedCellAnalyzer.cs

```diff
@@ -0,0 +1,205 @@
+        }
+
+        public ParsedCellAnalyzer(ExcelParser.ParsedCell cell)
+        {
+            isFunction = false;
+            analyzedCell = cell;
+        }
+
+
+        public static string Analyze(string formula, ExcelParser.ParsedCell cell = null)
+        {
+            var engine = new Engine(new PowerFxConfig());
+            var parseResult = engine.Parse(formula);
+            return Analyze(parseResult.Root, cell);
+        }
+
+        public static string Analyze(TexlNode node, ExcelParser.ParsedCell cell = null)
+        {
+            var analyzer = new ParsedCellAnalyzer(cell);
+            return node.Accept(analyzer, Precedence.None);
+        }
+
+
+        public override String Visit(UnaryOpNode node, Precedence context)
+        {
+            return node.Op.ToString() + node.Child.Accept(this, Precedence.None);
+        }
+
+        public override String Visit(BoolLitNode node, Precedence context)
+        {
+            return node.ToString();
+        }
+
+        public override String Visit(NumLitNode node, Precedence context)
+        {
+            if (isFunction && analyzedCell != null) // if number within a func call, treat it literally (temporary feature)
+            {
+                return node.ActualNumValue.ToString();
+            }
+            else
+            {
+                return Utils.CreateVariable(analyzedCell.SheetName, analyzedCell.CellId, node);
+            }
+        }
+
+        public override String Visit(FirstNameNode node, Precedence context)
+        {
+            if (isFunction && analyzedCell != null) // if FirstName within a func call, treat as cell reference and convert to generically named var (temporary feature)
+            {
+                return Utils.GenerateGenericName(analyzedCell.SheetName, node.Ident.Name.Value);
+            }
+            else // otherwise, treat it as a new String variable creation and a cell with text in it
+            {
+                return Utils.CreateVariable(analyzedCell.SheetName, analyzedCell.CellId, "\"" + node.Ident.Name.Value + "\"");
+            }
+        }
+
+        public override String Visit(StrLitNode node, Precedence context)
+        {
+            if (isFunction && analyzedCell != null) // if number within a func call, treat it literally (temporary feature)
+            {
+                return "\"" + node.Value + "\""; // wrap text in quotes so it is treated as literal text and not accidentally as variable
+            }
+            else
+            {
+                return Utils.CreateVariable(analyzedCell.SheetName, analyzedCell.CellId, node);
+            }
+        }
+
+        public override String Visit(BinaryOpNode node, Precedence context)
+        {
+            var binaryOpNode = (BinaryOpNode)node;
+
+            String left = node.Left.Accept(this, Precedence.None);
+            String right = node.Right.Accept(this, Precedence.None);
+
+            return left + " + " + right;
```

Comment:
should + be hard coded? Or should it be the Op value?

Reply:
Fixed this bug earlier - good catch!